### ISSUE TITLE
Mistake in TDS::Default_vertex_converter

### ISSUE THREAD

**Issue by tcommandeur** [CGAL/cgal#3709]
## Issue Details

When copying a Triangulation Data Structure 2 to another Triangulation Data Structure 2 the Default_vertex_converter definition is wrong, it returns `Vertex_src( src.point() );` instead of `Vertex_tgt( src.point() );`

## Source Code
```typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
typedef CGAL::Projection_traits_xy_3<K>                     Gt;
typedef CGAL::Exact_predicates_tag                          Itag;
typedef CGAL::Constrained_Delaunay_triangulation_2<Gt, CGAL::Default, Itag>	CDT;

struct FaceInfo2
{
  FaceInfo2() {}
  int number;
};
typedef CGAL::Triangulation_face_base_with_info_2<FaceInfo2, Gt>	Fbb;
typedef CGAL::Constrained_triangulation_face_base_2<Gt, Fbb>		Fb;
typedef CGAL::Triangulation_data_structure_2<Vb, Fb>			Tds;
typedef CGAL::Constrained_Delaunay_triangulation_2<Gt, Tds, Itag>	CDTinfo;

CDT cdt;
CDTinfor cdtinfo;
CGAL::internal::TDS_2::Default_vertex_converter<typename Tds::Vertex, CDT::Vertex> setv;
CGAL::internal::TDS_2::Default_face_converter<typename Tds::Face, CDT::Face>  setf;
cdt.tds().copy_tds(cdtinfo.tds(), cdtinfo.infinite_vertex(), setv, setf);
```

## Solution
https://github.com/CGAL/cgal/blob/master/TDS_2/include/CGAL/Triangulation_data_structure_2.h#L2022
Current :
```
//utilities for copy_tds
namespace internal { namespace TDS_2{
  template <class Vertex_src,class Vertex_tgt>
  struct Default_vertex_converter
  {
    Vertex_tgt operator()(const Vertex_src& src) const {
      return Vertex_src( src.point() );
    }
    
    void operator()(const Vertex_src&,Vertex_tgt&) const {}
  };
```

Correct:
```
//utilities for copy_tds
namespace internal { namespace TDS_2{
  template <class Vertex_src,class Vertex_tgt>
  struct Default_vertex_converter
  {
    Vertex_tgt operator()(const Vertex_src& src) const {
      return Vertex_tgt( src.point() );
    }
    
    void operator()(const Vertex_src&,Vertex_tgt&) const {}
  };
```
## Environment

* Operating system (Windows 64 bits):
* Compiler: MSVC 2017
* Release or debug mode: both
* Specific flags used (if any):
* CGAL version: 4.12
* Boost version: 1.68
* Other libraries versions if used (Eigen, TBB, etc.):

**Comment by afabri** [comment]
I confirm your observation and fix, but I have several remarks
* You should not  use something from namespace ` CGAL::internal`, as it might be removed, renamed...
* Don't put code inside an issue but put it on gist.github.com.   Your code above has typos (`CDTInfor`), and I am pretty sure that it does not even compile with your suggested fix, because your two triangulation types use 2D and 3D points respectively.

**Comment by lrineau** [comment]
Fixed by #3725. Will be in CGAL-4.13.1 and CGAL-4.14. Sorry for CGAL-4.12.x: Andreas did not base his PR on the branch for CGAL-4.12.2.
